build/main.rs

```diff
@@ -52,15 +53,25 @@ fn main() {
         println!("Failed to run rustfmt, will skip formatting generated files.")
     }
 
+    // Check if cbindgen is installed
+    let has_cbindgen = Command::new("cbindgen")
+        .arg("--version")
+        .stdin(Stdio::inherit())
+        .status()
+        .is_ok();
+
+    if !has_cbindgen {
+        println!("Failed to run cbindgen, will skip generating C bindings.")
+    }
+
```

Comment:
You already added it to the build-dependencies in the Cargo.toml.
So you should use it as a libray i.e. use <https://docs.rs/cbindgen/latest/cbindgen/struct.Builder.html>, rather than calling cbindgen as an external command.

Reply:
Something like
```suggestion
println!("cargo:rerun-if-changed=cbindgen.toml");
let manifest_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
cbindgen::generate(&manifest_dir ).unwrap().write_to_file("path/to/file.h")
```

Reply:
Note that https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script states that 

> Build scripts may save any output files or intermediate artifacts in the directory specified in the [OUT_DIR environment variable](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts). Scripts should not modify any files outside of that directory.

Given that its only a `should` and not a `must` I think its reasonable to ignore this here.

Reply:
you probably don't want to actually have the second unwarp as a syntax error in any file would cause cbindgen to fail, which would cause the build script to fail, which would cause the build to fail, likely breaking syntax highlighting etc. making finding and fixing the problem difficult.